cli/proc_state.rs

```diff
@@ -164,15 +164,16 @@ impl ProcState {
       eprintln!("{}", colors::yellow(msg));
     }
 
-    let cache_usage = if flags.cached_only {
-      CacheSetting::Only
-    } else if !flags.cache_blocklist.is_empty() {
-      CacheSetting::ReloadSome(flags.cache_blocklist.clone())
-    } else if flags.reload {
-      CacheSetting::ReloadAll
-    } else {
-      CacheSetting::Use
-    };
+    let cache_usage =
+      if flags.cached_only || env::var("DENO_CACHED_ONLY").is_ok() {
```

Comment:
It just being set is not good enough I think. We should also consider if it is set to false as it being unset. Maybe error if it is a value other than "true" or "false" or "" (the last being an alias for "false")

Reply:
this is uncommon behavior - most unix applications only test for it being set

Reply:
@piscisaureus Do you have an opinion?

Reply:
> most unix applications only test for it being set

I'd argue that even if it is unix convention, `DENO_CACHED_ONLY=false` meaning the same as `DENO_CACHED_ONLY=true` is unexpected to many. I think we can just error when the value of `DENO_CACHED_ONLY` is anything other than `1` or `true`.

Reply:
But why would anyone ever do `DENO_CACHED_ONLY=false`? It's silly. This is a perennial argument we have.

Sure, I'll implement this non-standard behavior. But it's silly. I assure you binary env vars in Golang do not do this.

Reply:
> But why would anyone ever do DENO_CACHED_ONLY=false?

It's not unlikely someone would do this I'd think. I also don't see the harm about just enforcing one specific value rather than just allowing all. Being more strict will catch accidental issues early.

```js
const useCachedOnly: boolean = ...;

Deno.run({
  cmd: ["deno", ...],
  env: {
    "DENO_CACHED_ONLY": String(useCachedOnly)
  }
})
```

Reply:
Updated. Is this ok? Or do we also need to suppose "DENO_CACHED_ONLY=False" ? What about "DENO_CACHED_ONLY=No" ?

Reply:
Looks ok. But I would have also been happy with just parsing `DENO_CACHED_ONLY=1` as `true` and erroring for all other cases :-)

Reply:
@piscisaureus has said in the past that we need to also support "yes", even for very internal env vars like `V8_FORCE_DEBUG`. 

https://github.com/denoland/rusty_v8/blob/845aced92cc1816dda248c882bfb1ec6c4dca81a/build.rs#L249-L254

@piscisaureus do we need to support "yes" here?

Reply:
> @piscisaureus Do you have an opinion?

I agree with "" and "0" and "false" meaning the flag is disabled.